test/modern_bpf/test_suites/syscall_exit_suite/accept_x.cpp

```diff
@@ -0,0 +1,343 @@
+	}
+	assert_syscall_state(SYSCALL_SUCCESS, "bind (server)", syscall(__NR_bind, server_socket_fd, (struct sockaddr*)&server_addr, sizeof(server_addr)), NOT_EQUAL, -1);
+	assert_syscall_state(SYSCALL_SUCCESS, "listen (server)", syscall(__NR_listen, server_socket_fd, QUEUE_LENGTH), NOT_EQUAL, -1);
+
+	/* The server now is ready, we need to create at least one connection from the client. */
+	int32_t client_socket_fd = syscall(__NR_socket, AF_UNIX, SOCK_STREAM, 0);
+	assert_syscall_state(SYSCALL_SUCCESS, "socket (client)", client_socket_fd, NOT_EQUAL, -1);
+
+	/* We need to bind the client socket with an address otherwise we cannot assert against it. */
+	struct sockaddr_un client_addr;
+	bzero(&client_addr, sizeof(client_addr));
+	client_addr.sun_family = AF_UNIX;
+	if(strncpy(client_addr.sun_path, UNIX_CLIENT, MAX_SUN_PATH) == NULL)
+	{
+		FAIL() << "'strncpy (client)' must not fail." << std::endl;
+	}
+	assert_syscall_state(SYSCALL_SUCCESS, "bind (client)", syscall(__NR_bind, client_socket_fd, (struct sockaddr*)&client_addr, sizeof(client_addr)), NOT_EQUAL, -1);
+	assert_syscall_state(SYSCALL_SUCCESS, "connect (client)", syscall(__NR_connect, client_socket_fd, (struct sockaddr*)&server_addr, sizeof(server_addr)), NOT_EQUAL, -1);
+
+	/* We don't want to get any info about the connected socket so `addr` and `addrlen` are NULL. */
+	int connected_socket_fd = syscall(__NR_accept, server_socket_fd, NULL, NULL);
+	assert_syscall_state(SYSCALL_SUCCESS, "accept (server)", connected_socket_fd, NOT_EQUAL, -1);
+
+	/* Cleaning phase */
+	syscall(__NR_shutdown, connected_socket_fd, 2);
+	syscall(__NR_shutdown, server_socket_fd, 2);
+	syscall(__NR_shutdown, client_socket_fd, 2);
+	syscall(__NR_close, connected_socket_fd);
+	syscall(__NR_close, server_socket_fd);
+	syscall(__NR_close, client_socket_fd);
+	syscall(__NR_unlinkat, 0, UNIX_CLIENT, 0);
+	syscall(__NR_unlinkat, 0, UNIX_SERVER, 0);
+
+	/*=============================== TRIGGER SYSCALL ===========================*/
+
+	evt_test->disable_capture();
+
+	evt_test->assert_event_presence();
+
+	if(HasFatalFailure())
+	{
+		return;
+	}
+
+	evt_test->parse_event();
+
+	evt_test->assert_header();
+
+	/*=============================== ASSERT PARAMETERS  ===========================*/
+
+	/* Parameter 1: fd (type: PT_FD) */
+	evt_test->assert_numeric_param(1, (int64_t)connected_socket_fd);
+
+	/* Parameter 2: tuple (type: PT_SOCKTUPLE) */
+	/* The server performs an `accept` so the `client` is the src. */
+	evt_test->assert_tuple_unix_param(2, PPM_AF_UNIX, UNIX_SERVER);
+
+	/* Parameter 3: queuepct (type: PT_UINT8) */
+	/* we expect 0 elements in the queue so 0%. */
+	evt_test->assert_numeric_param(3, (uint8_t)0);
+
+	/* Parameter 4: queuelen (type: PT_UINT32) */
+	/* we expect 0 elements. */
+	evt_test->assert_numeric_param(4, (uint32_t)0);
+
+	/* Parameter 5: queuemax (type: PT_UINT32) */
+	/* In unix sockets the maximum queue length seems to be 512. */
+	uint32_t unix_max_queue_len = 512;
```

Comment:
Seems to relate to the number in /proc/sys/net/unix/max_dgram_qlen (default).

Reply:
I know this is really strange but i don't know why `listen()` seems to not set it :/

Reply:
i've tried a little bit but i've not understood the reason behind this :/  BTW now I extract the information from `/proc/sys/net/unix/max_dgram_qlen` so the test should never fail :) Thank you for the suggestion

Reply:
Thanks a lot! Not sure to open an issue to track it? wdyt?

Reply:
yep i've added it to our tracking issue under `accept`:arrow_right: